src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTime.cs

```diff
@@ -10,6 +10,19 @@ namespace System.Text.Json
 {
     public sealed partial class Utf8JsonWriter
     {
+        public void WriteString(JsonEncodedText propertyName, DateTime value)
+            => WriteStringHelper(propertyName.EncodedUtf8Bytes, value);
+
+        private void WriteStringHelper(ReadOnlySpan<byte> utf8PropertyName, DateTime value)
+        {
+            Debug.Assert(utf8PropertyName.Length <= JsonConstants.MaxTokenSize);
```

Comment:
This isn't a valid assert.  Nothing stopped a JsonEncodedText from being the entirety of War and Peace.

Reply:
(OK, so apparently MaxTokenSize is ~358 million, and War and Peace is under 600 thousand words -- so about 1.2 million characters (5 letters per word, plus spaces, punctuation fits in the rounding-up-to-5).  But a mere 32 copies of [Artamène ou le Grand Cyrus](https://en.wikipedia.org/wiki/Artam%C3%A8ne) (possibly actually the longest book ever) would exceed that value limit.

So it's not realistically going to be exceeded, but it's not really provably not exceeded, and you can trivially exceed it with the UTF8 payload `new byte[400_000_000].AsSpan().Fill((byte)'a')`, unless there was a limit I missed somewhere.)

Reply:
`JsonEncodedText` doesn't allow `> ~358 million` which is why the assert holds true here.

Reply:
It looks (to me) like `JsonEncodedText.Encode(new string('+', 60_000_000)) will violate the assert.

60M characters is less than 238M characters, so it clears JsonWriterHelper.ValidateValue.  Then it goes into the escaping routine and each `+` turns into `\u002b`, meaning there are now 360M bytes, which exceeds 358M bytes, so it exceeds this assert.

Reply:
> so it exceeds this assert.

You are absolutely right. The assert is constraining it too much. The un-escaped limit is 358M (what `JsonEncodedText` checks for). Once encoded, it can safely expand up to 2GB (6x factor). Asserting 358M isn't valid (60M escaped '+' is allowed). Will fix the asserts.